### PR TITLE
Better CI: clj-kondo, cljfmt

### PR DESCRIPTION
**Pull request by kamilchulakov**

### CODE REVIEW

**Comment by kamilchulakov** [comment]
- Can't use actions without Dockerfile builds (DeLaGuardo/setup-clj-kondo fails with Unable to process command '::add-path::/opt/hostedtoolcache/CljKondo/2020.04.05/x64' successfully.)
- Need to run workflow to check if Dockerfiles don't cause "docker build failed" with permission denied

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -29,6 +29,17 @@ jobs:
...
           cli: 1.11.1.1155
+          
+      - name: clj-kondo checks
+        uses: DeLaGuardo/clojure-lint-action@v1
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          
+      - uses: bltavares/actions/cljfmt@1.0.8
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (ryukzak) [inline .github/workflows/main.yml:40]: Why you need token here?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -29,6 +29,17 @@ jobs:
...
+        with:
+          args: autofix src test
```
Comment (ryukzak) [inline .github/workflows/main.yml:42]: Only check

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -29,6 +29,17 @@ jobs:
         uses: DeLaGuardo/setup-clojure@9.5
         with:
           cli: 1.11.1.1155
+          
+      - name: clj-kondo checks
+        uses: DeLaGuardo/clojure-lint-action@v1
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (ryukzak) [inline .github/workflows/main.yml:36]: Why you need token here?

**Review by ryukzak — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml (hunk 1 above)
Comment (kamilchulakov) [reply .github/workflows/main.yml:40]: token is needed for fix commit

File: .github/workflows/main.yml (hunk 3 above)
Comment (kamilchulakov) [reply .github/workflows/main.yml:36]: > secrets.GITHUB_TOKEN is needed here
> to publish annotations back to github

File: .github/workflows/main.yml (hunk 2 above)
Comment (kamilchulakov) [reply .github/workflows/main.yml:42]: ok

File: .github/workflows/main.yml (hunk 3 above)
Comment (kamilchulakov) [reply .github/workflows/main.yml:36]: I'll use another action option which doesn't require it

File: .github/workflows/main.yml (hunk 1 above)
Comment (kamilchulakov) [reply .github/workflows/main.yml]: don't use this action anymore

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -29,6 +29,17 @@ jobs:
...
+      - name: Setup clj-kondo
+        uses: DeLaGuardo/setup-clj-kondo@master
+        with:
+          version: '2021.12.16'
+          
+      - name: clj-kondo lint
+        run: clj-kondo --lint deps.edn --config '{:output {:pattern "::{{level}} file={{filename}},line={{row}},col={{col}}::{{message}}"}}'
+          
+      - name: cljfmt check
+        run: clojure -Sdeps '{:deps {cljfmt {:mvn/version "0.9.2"}}}' -M -m cljfmt.main check
```
Comment (ryukzak) [inline .github/workflows/main.yml:42]: Add these dependency to the project as development. It looks like a more narrow way to do this.

**Review by ryukzak — CHANGES_REQUESTED** [review]

**Comment by ryukzak** [comment]
From the terminal:

```console
~/src/course-bot master *1 ❯ clj-kondo --lint src --lint test                                                                                                                                   ✘ 3
src/course_bot/core.clj:3:14: warning: namespace clojure.data.csv is required but never used
src/course_bot/core.clj:5:14: warning: namespace clojure.java.io is required but never used
src/course_bot/core.clj:6:14: warning: namespace course-bot.dialog is required but never used
src/course_bot/core.clj:8:14: warning: namespace course-bot.quiz is required but never used
src/course_bot/core.clj:11:14: warning: namespace course-bot.presentation is required but never used
src/course_bot/core.clj:16:14: warning: namespace clojure.string is required but never used
src/course_bot/core.clj:17:14: warning: namespace clojure.pprint is required but never used
src/course_bot/core.clj:17:37: warning: #'clojure.pprint/pprint is referred but never used
src/course_bot/core.clj:57:4: warning: Unresolved var: essay/essay-talk
src/course_bot/core.clj:58:4: warning: Unresolved var: essay/assign-essay-talk
src/course_bot/core.clj:59:4: warning: Unresolved var: essay/essay-review-talk
src/course_bot/core.clj:60:4: warning: Unresolved var: essay/essay-status-talk
src/course_bot/core.clj:61:4: warning: Unresolved var: essay/essay-results-talk
src/course_bot/core.clj:62:4: warning: Unresolved var: essay/essays-without-review-talk
src/course_bot/core.clj:117:30: error: Unresolved symbol: tx
src/course_bot/core.clj:122:6: warning: unused binding args
src/course_bot/csa.clj:97:14: warning: unused binding args
src/course_bot/essay.clj:223:42: warning: unused binding conf
...
```

File: src/course_bot/core.clj [hunk 5]
```diff
@@ -1,20 +1,13 @@
 (ns course-bot.core
-  (:require [codax.core :as c]
```
Comment (ryukzak) [inline src/course_bot/core.clj:2]: Please, remove this legacy file.

File: src/course_bot/presentation.clj [hunk 6]
```diff
@@ -57,7 +57,7 @@
           group (-> data (get id) :presentation (get pres-key) :group)
           lessons (-> data :presentation (get pres-key) (get group))]
       (->> lessons
-           (filter (fn [[_k v]] (-> v :stud-ids empty? not)))
+           (filter (fn [[_k v]] (-> v :stud-ids seq)))
```
Comment (ryukzak) [inline src/course_bot/presentation.clj:60]: About this suggestion. Revert your fix and suppress that warning. From my point of view -- it is very bad practice.

Here and in other places.

File: src/course_bot/talk.clj [hunk 7]
```diff
@@ -43,7 +43,7 @@
       (keyword (first args))
       nil)))
 
-(defn id-from-arg [tx text]
+(defn id-from-arg [text]
```
Comment (ryukzak) [inline src/course_bot/talk.clj:46]: If I don't mistake, we don't use this function. Remove it.

**Review by ryukzak — CHANGES_REQUESTED** [review]
Fix CI. Fix merge conflict by rebase.

File: src/course_bot/talk.clj (hunk 7 above)
Comment (kamilchulakov) [reply src/course_bot/talk.clj:46]: Yeah, it is unused

File: src/course_bot/presentation.clj (hunk 6 above)
Comment (kamilchulakov) [reply src/course_bot/presentation.clj]: Suppressed such warning in config

**Comment by kamilchulakov** [comment]
> Fix CI. Fix merge conflict by rebase.

Is rebase preferred over merge?

**Comment by kamilchulakov** [comment]
squash + rebase instead of merge => approve?

**Pull request merged**